tests/simpletest.py

```diff
@@ -277,6 +277,49 @@ def test_intersection_without_copy_template(self):
                 self.assertTrue(all(i in bf2 for i in range(50, 150)))
                 self.assertTrue(all(i not in bf2 for i in range(150, 200)))
 
+    def test_write_operation_on_readonly_file_raises_exception(self):
+        with tempfile.NamedTemporaryFile() as tmp1:
+            with tempfile.NamedTemporaryFile() as tmp2:
+                pybloomfilter.BloomFilter(1000, 0.01, tmp1.name)
+                bf1 = pybloomfilter.BloomFilter.open(tmp1.name)
+
+                bf2 = bf1.copy_template(tmp2.name)
+                bf2.add('bf2')
+
+                with self.assertRaises(ValueError):
+                    bf1.clear_all()
+
+                with self.assertRaises(ValueError):
+                    bf1.add('test')
+
+                with self.assertRaises(ValueError):
+                    bf1 |= bf2
+
+                with self.assertRaises(ValueError):
+                    bf1 &= bf2
+
+                with self.assertRaises(ValueError):
+                    bf1.union(bf2)
+
+                with self.assertRaises(ValueError):
+                    bf1.intersection(bf2)
+
+    def test_write_operation_on_writable_file_does_not_raise_exception(self):
```

Comment:
I would add a test here where one opens the bf like the previous test, but with writable mode, in order to check that the bf can be modified without ValueError or segfault

Reply:
I added it to the same test since they are basically the same